tests/functional/conftest.py

```diff
@@ -78,102 +60,7 @@ def event_loop():
             await asyncio.sleep(1)
 
 
-@pytest.fixture
-async def get_app(model):
-    '''Returns the application requested'''
-    async def _get_app(name):
-        try:
-            return model.applications[name]
-        except KeyError:
-            raise JujuError("Cannot find application {}".format(name))
-    return _get_app
-
-
-@pytest.fixture
-async def get_unit(model):
-    '''Returns the requested <app_name>/<unit_number> unit'''
-    async def _get_unit(name):
-        try:
-            (app_name, unit_number) = name.split('/')
-            return model.applications[app_name].units[unit_number]
-        except (KeyError, ValueError):
-            raise JujuError("Cannot find unit {}".format(name))
-    return _get_unit
-
-
-@pytest.fixture
-async def get_entity(get_unit, get_app):
-    '''Returns a unit or an application'''
-    async def _get_entity(name):
-        try:
-            return await get_unit(name)
-        except JujuError:
-            try:
-                return await get_app(name)
-            except JujuError:
-                raise JujuError("Cannot find entity {}".format(name))
-    return _get_entity
-
-
-@pytest.fixture
-async def run_command(get_unit):
-    '''
-    Runs a command on a unit.
-
-    :param cmd: Command to be run
-    :param target: Unit object or unit name string
-    '''
-    async def _run_command(cmd, target):
-        unit = (
-            target
-            if isinstance(target, juju.unit.Unit)
-            else await get_unit(target)
-        )
-        action = await unit.run(cmd)
-        return action.results
-    return _run_command
-
-
-@pytest.fixture
-async def file_stat(run_command):
-    '''
-    Runs stat on a file
-
-    :param path: File path
-    :param target: Unit object or unit name string
-    '''
-    async def _file_stat(path, target):
-        cmd = STAT_CMD % path
-        results = await run_command(cmd, target)
-        return json.loads(results['Stdout'])
-    return _file_stat
-
-
-@pytest.fixture
-async def file_contents(run_command):
-    '''
-    Returns the contents of a file
-
-    :param path: File path
-    :param target: Unit object or unit name string
-    '''
-    async def _file_contents(path, target):
-        cmd = 'cat {}'.format(path)
-        results = await run_command(cmd, target)
-        return results['Stdout']
-    return _file_contents
-
-
-@pytest.fixture
-async def reconfigure_app(get_app, model):
-    '''Applies a different config to the requested app'''
-    async def _reconfigure_app(cfg, target):
-        application = (
-            target
-            if isinstance(target, juju.application.Application)
-            else await get_app(target)
-        )
-        await application.set_config(cfg)
-        await application.get_config()
-        await model.block_until(lambda: application.status == 'active')
-    return _reconfigure_app
+@pytest.fixture(scope='module')
+async def utils(controller, model):
```

Comment:
Should we just call this 'juju' or 'jujutools' or something to make it a bit clearer in tests what it is interacting with?

Reply:
Yup good point I'll change that